Content.Server/Atmos/AtmosDirection.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using Robust.Shared.Maths;
+
+namespace Content.Server.Atmos
+{
+    /// <summary>
+    ///     The reason we use this over <see cref="Direction"/> is that we are going to do some heavy bitflag usage.
+    /// </summary>
+    [Flags]
+    public enum AtmosDirection : byte
+    {
+        Invalid = 0,
+        North   = 1 << 0,
+        South   = 1 << 1,
+        East    = 1 << 2,
+        West    = 1 << 3,
+
+        NorthEast = North | East,
+        NorthWest = North | West,
+        SouthEast = South | East,
+        SouthWest = South | West,
+
+        All = North | South | East | West,
+    }
+
+    public static class AtmosDirectionHelpers
+    {
+        public static AtmosDirection GetOpposite(this AtmosDirection direction)
+        {
+            return direction switch
+            {
+                AtmosDirection.North => AtmosDirection.South,
+                AtmosDirection.South => AtmosDirection.North,
+                AtmosDirection.East => AtmosDirection.West,
+                AtmosDirection.West => AtmosDirection.East,
+                AtmosDirection.NorthEast => AtmosDirection.SouthWest,
+                AtmosDirection.NorthWest => AtmosDirection.SouthEast,
+                AtmosDirection.SouthEast => AtmosDirection.NorthWest,
+                AtmosDirection.SouthWest => AtmosDirection.NorthEast,
+                _ => AtmosDirection.Invalid
```

Comment:
Would it be better to throw an exception directly in this and other similar cases? It would prevent an invalid direction from potentially propagating unknown. It would also follow the same behavior that `TileAtmosInfo` does when accessed by index.

Reply:
You can then also remove the invalid check in TileAtmosphere line 625, which removes the assumption of there being only one invalid direction.

Reply:
Same goes for line 825 and pretty much all other checks of direction invalid.

Reply:
Keep in mind, some of the direction checks are *actually* important, since `TileAtmosInfo` defaults the direction to Invalid. I removed the ones I felt that were unnecessary